parser-java/kvs-archived-media-parallel-processing-image-rekognition-sample/src/main/java/com/amazonaws/kinesisvideo/KinesisVideoArchivedParallelProcessingExample.java

```diff
@@ -0,0 +1,155 @@
+        this.executorService = Executors.newFixedThreadPool(threads);
+        this.startTimestamp = startTimestamp;
+        this.endTimestamp = endTimestamp;
+        this.sampleRate = sampleRate;
+        this.tasks = tasks;
+    }
+
+    public void execute() throws InterruptedException, IOException, ParseException, ExecutionException {
+
+        List<TimestampRange> timestampRanges = partitionTimeRange(startTimestamp, endTimestamp);
+
+        String listFragmentsEndpoint = getListFragmentsEndpoint(getStreamName());
+        String getMediaFragmentListEndpoint = getGetMediaForFragmentListEndpoint(getStreamName());
+
+        for (TimestampRange timestampRange: timestampRanges) {
+
+            log.info(timestampRange.toString());
+
+            FrameVisitor frameVisitor = FrameVisitor.create(H264FrameLabelDetector.create(sampleRate), Optional.empty(), Optional.of(1L));
+
+            GetMediaArchivedRekognitionWorker getMediaArchivedRekognitionWorker = GetMediaArchivedRekognitionWorker.create(getStreamName(),
+                    getCredentialsProvider(),
+                    getRegion(),
+                    streamOps.getAmazonKinesisVideo(),
+                    new FragmentSelector()
+                            .withFragmentSelectorType(FragmentSelectorType.SERVER_TIMESTAMP)
+                            .withTimestampRange(timestampRange),
+                    frameVisitor,
+                    listFragmentsEndpoint,
+                    getMediaFragmentListEndpoint);
+
+            executorService.submit(getMediaArchivedRekognitionWorker);
+
+        }
+
+        //Wait for the workers to finish.
+        executorService.shutdown();
+        executorService.awaitTermination(awaitTerminationTime, TimeUnit.SECONDS);
+        if (!executorService.isTerminated()) {
+            log.warn("Shutting down executor service by force");
+            executorService.shutdownNow();
+        } else {
+            log.info("Executor service is shutdown");
+        }
+    }
+
+    /* Create N time stamp ranges so that each of the N threads can call ListFragments on a specified partition */
+    public List<TimestampRange> partitionTimeRange(String startTimestamp, String endTimestamp) throws ParseException{
+        List<TimestampRange> timestampRanges= new ArrayList<>();
+
+        try {
+            Date startDate = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss").parse(startTimestamp);
+            Date endDate = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss").parse(endTimestamp);
+
+            /* Time between two timestamps in milliseconds */
+            long timespan = (endDate.getTime() - startDate.getTime());
+            long task_timeslice = timespan / tasks;
+
+            long startTime = startDate.getTime();
+            long endTime = endDate.getTime();
+
+            long taskStart = startTime;
+            long taskEnd = startTime + task_timeslice;
+
+            while (taskStart <= endTime) {
```

Comment:
This can end up with an out of bound range. Let say we have this imaginary example:

```
startTime = 0;
endTime = 15;
tasks = 3;
taskTimeslice = (endTime - startTime) / tasks; // equal 5

taskStart = 0;
taskEnd = 5;
```

|Iteration|taskStart|taskEnd|
|-|-|-|
|1|0|5|
|2|6|11|
|3|12|17| 

After 3 iterations, the `taskEnd` goes beyond 15 (`endTime`)